src/test.h

```diff
@@ -144,41 +144,41 @@ struct VirtualFileSystem : public DiskInterface {
   }
 
   // DiskInterface
-  virtual TimeStamp Stat(const string& path, string* err) const;
-  virtual bool WriteFile(const string& path, const string& contents);
-  virtual bool MakeDir(const string& path);
-  virtual Status ReadFile(const string& path, string* contents, string* err);
-  virtual int RemoveFile(const string& path);
+  virtual TimeStamp Stat(const std::string& path, std::string* err) const;
+  virtual bool WriteFile(const std::string& path, const std::string& contents);
+  virtual bool MakeDir(const std::string& path);
+  virtual Status ReadFile(const std::string& path, std::string* contents, std::string* err);
```

Comment:
Can you clang-format this line (and the others with the same issue) so that it stays in the 80 columns limit?

Reply:
Yes. I will do so.

Reply:
Clang formatting the code (using the supplied .clang-format file) results in *lots* of changes.

I suppose I'll edit things by hand then.

Reply:
Actually.

Can we leave formatting changes, such as the 80 column limit, to a separate PR?

Reply:
Use Visual Studio Code (with C/C++ or ClangFormat extension). You can open the diff of each file and select the changed lines (on the right). Right-click on the selection and select "Format selection". It will only format the selected line then.

You can also add rulers so that you know where you're over the 80 limit to only format those.

Reply:
I suspect you were writing your reply while I asked:

Can we leave formatting changes, such as the 80 column limit, to a separate PR?

Reply:
I suggest that .clang-format be run on the whole codebase after https://github.com/ninja-build/ninja/milestones/1.10.0

Reply:
No, sorry. I would like to avoid line touches just for formatting.

Reply:
You understand that the diff between what we have, and what clang-format gives us is over 5,000 lines right?

clang-format -i src/*.h src/*.cc ; git diff | grep -e "+" -e "-" |  wc -l
8808

You need to rip the bandaid off eventually.

It would be better for *everyone* if we just applied clang format, and moved on.

Reply:
I disagree.

Reply:
The second commit on this PR addresses lines changed by the first commit that now exceed 80 columns.

Reply:
> I disagree.

Can you please provide explanations in the future? No meaningful discussion can be had if one party provides explanations for their position, e.g. The clang formatting rules supplied with the project cause over 5 thousand lines of changes

when the other person just says they disagree and that's all.

Reply:
You didn't explain, why it would be better for everyone, if we just applied clang-format and moved on, either. That's why I felt the need to state, that I disagree with that opinion.

Reply:
Ok, I can see how you thought that.

I thought it was implied by "There are over 5 thousand lines of changes", and the work that contributors have to do by hand in order to comply with the not-in-clang-format rules that the project has.

E.g., it took me 20+ minutes to go line by line and correct the overly long lines in a diff viewer.

Why bother to offer a clang format ruleset if it's not used?

If we apply the rules to the entire repository, and then consistently enforce the use of clang format for any PR that's merged (either by requiring the PR be formatted with clang format prior to merging, or applying clang format immediately after merging), the pain of requiring each contributor to comply individually is condensed into a single "flag day", and then goes away after that.

Reply:
Would have took you only a few minutes if you done it with Visual Studio Code.

> Why bother to offer a clang format ruleset if it's not used?

There's not clang-format ruleset. If in doubt -> use clang-format.

Reply:
> Would have took you only a few minutes if you done it with Visual Studio Code.

This would require:
1) I have Visual Studio Code, or am able to install it. You don't know what operating system I use, or if it's practical for me to download and install an arbitrary program.
2) That Visual Studio Code is, in fact, faster to use than whatever tool I'm already familiar with.

> There's not clang-format ruleset. If in doubt -> use clang-format.

.clang-format == "clang-format ruleset"

Reply:
> That Visual Studio Code is, in fact, faster to use than whatever tool I'm already familiar with.

Apparently it is, otherwise it wouldn't have took you 20 minutes.

Reply:
I did a lot more than just re-flow 80+column lines. I compared the clang formatting of every line in the header files to make sure it all looked right.

Visual studio code wouldn't have helped much with that.

Reply:
Ah I see. No need to do that :)